keel-igor/src/main/kotlin/com/netflix/spinnaker/keel/services/ArtifactMetadataService.kt

```diff
@@ -0,0 +1,36 @@
+package com.netflix.spinnaker.keel.services
+
+import com.netflix.spinnaker.igor.BuildService
+import com.netflix.spinnaker.keel.api.artifacts.ArtifactMetadata
+import com.netflix.spinnaker.keel.api.artifacts.BuildMetadata
+import com.netflix.spinnaker.keel.api.artifacts.GitMetadata
+import kotlinx.coroutines.runBlocking
+import org.springframework.stereotype.Component
+
+/**
+ * Provides functionality to convert build metadata, which is coming from internal service, to artfiact metadata (via igor).
+ */
+@Component
+class ArtifactMetadataService(
+  private val buildService: BuildService
+) {
+
+  fun getArtifactMetadata(
+    commitId: String?,
+    buildNumber: String?
+  ): ArtifactMetadata? {
+
+    if (commitId == null || buildNumber == null) {
+      return null
+    }
+      val builds = runBlocking {
+        buildService.getArtifactMetadata(commitId = commitId, buildNumber = buildNumber)
```

Comment:
retries might be good here, as well as error handling

Reply:
+1. Retries on `50x` and fail gracefully on `404`.